WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java

```diff
@@ -2316,9 +2320,51 @@ private GutenbergPropsBuilder getGutenbergPropsBuilder() {
         );
     }
 
-    private Boolean canViewEditorOnboarding() {
-        return (mAccountStore.getAccount().getUserId() % 100 >= (100 - EDITOR_ONBOARDING_PHASE_PERCENTAGE)
-                || BuildConfig.DEBUG);
+    /**
+     * Temporary method for the Editor Onboarding project to control the percentage
+     * of users able to use this new editor onboarding tooltip feature. This will eventually
+     * be removed when the functionality is opened to all users.
+     */
+    private boolean canViewEditorOnboarding() {
+        // Get the userId for the logged in user. If not WPcom, use the anonymous userId
+        // generated by tracks.
+        long userAccountId = mAccountStore.getAccount().getUserId();
+        String userId = String.valueOf(userAccountId);
+        if (userAccountId == 0) {
+            SharedPreferences preferences = PreferenceManager.getDefaultSharedPreferences(getApplicationContext());
+            userId = preferences.getString("nosara_tracks_anon_id", null);
+        }
+
+        int hashedUserId = convertUserIdToHash(userId, "can_view_editor_onboarding");
+        return hashedUserId % 100 >= (100 - EDITOR_ONBOARDING_PHASE_PERCENTAGE) || BuildConfig.DEBUG;
+    }
+
+    /**
+     * Temporary method for the Editor Onboarding project to control the percentage
+     * of users able to use this new editor onboarding tooltip feature. This will eventually
+     * be removed when the functionality is opened to all users.
+     *
+     * Uses cryptography to hash a userId plus a seed containing the feature flag, converts
+     * the result into a BigInteger, then divides that value by 100 and returns the remainder.
+     * The value returned will be used to determine if a user should see the editor onboarding
+     * tooltip.
+     *
+     * @param userId A unique ID for the logged in user.
+     * @param seed The feature string to keep hashed value results between features unique.
+     * @return An int between 1 and 100
+     */
+    private int convertUserIdToHash(String userId, String seed) {
```

Comment:
Somewhat related question, instead of passing this as a parameter, would it be simpler to instead just define the seed/salt inside of this method? The fact that it is being passed in makes me think that this is something we're expecting to change which, if I'm understanding correctly, we actually don't want or need the seed/salt to change, even if we were to use this with multiple different features.

Reply:
Regarding David's naming question, do we even really need a salt here? I thought that was only necessary to increase how easy it would be to reverse the hashing algorithm (i.e., it makes the cryptography more secure), but in this case we don't care if it is easy to decode it, so I'm wondering if we could simplify this by dropping the salt. 

I do not feel strongly about this, so it's not a blocker as far as I'm concerned.

Reply:
> Do we need a salt here? 

@mchowning it was my understanding that including the salt is the foundation of the idea to avoid including the same set of user for user-ID-based calculations. Without a unique salt, which we are using the experiment name, we would likely be right back to including the same set of users for any calculation across various experiments/rollouts. 

> Somewhat related question, instead of passing this as a parameter, would it be simpler to instead just define the seed/salt inside of this method? The fact that it is being passed in makes me think that this is something we're expecting to change which, if I'm understanding correctly, we actually don't want or need the seed/salt to change, even if we were to use this with multiple different features.

My understanding is that the salt would have to change if we used this approach for a different experiment. If it did not, then all the hashes would be the same between the two experiments, meaning the same set of users would be included for both experiments.

Reply:
👍   You're right. Thanks for explaining.

Reply:
Thanks @mchowning and @dcalhoun for the feedback - I really appreciate it. Since the hashing part is a temporary measure for a project on a tight timeline I'm good leaving this as-is, but will definitely keep all this in mind for the future. 👍